walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  private final Collection<TimeRange> availableTimes = new ArrayList<>();
+  private long meetingDuration;
+
+  /* Returns all available slots for a meeting to happen */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getAttendees().isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if (request.getDuration() >= TimeRange.WHOLE_DAY.duration() + 1) {
```

Comment:
Maybe this can be shortened by doing
```suggestion
    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
```